clash-lib/src/Clash/Core/Util.hs

```diff
@@ -789,11 +789,14 @@ isClockOrReset
   :: TyConMap
   -> Type
   -> Bool
-isClockOrReset m (coreView1 m -> Just ty)    = isClockOrReset m ty
-isClockOrReset _ (tyView -> TyConApp tcNm _) = case nameOcc tcNm of
+isClockOrReset m ty0@(tyView -> TyConApp tcNm _) = case nameOcc tcNm of
```

Comment:
`isClockishOrResetishOrEnableish` :wink:

But I do think this should be renamed to something more accurate, e.g. `isSpecialSignal`

Reply:
It does hurt a little, but I think `isClockOrResetOrEnable` is the best name here.

Reply:
@alex-mckenna You're in luck! I had to split out the function into `isClockOrReset` and `isEnable` anyway ;).